imagekit-sdk/src/test/java/io/imagekit/sdk/ImageKitTest.java

```diff
@@ -2,629 +2,1890 @@
+		MockWebServer server = new MockWebServer();
+		String responseJson = "{\n" + "    \"type\": \"file\",\n" + "    \"name\": \"sample-image11_g1Qv0wpqP.jpg\",\n"
+				+ "    \"createdAt\": \"2022-06-20T05:00:41.830Z\",\n"
+				+ "    \"updatedAt\": \"2022-06-20T05:00:43.263Z\",\n"
+				+ "    \"fileId\": \"62affef97db937b028f3b47a\",\n" + "    \"tags\": [\n" + "        \"Software\",\n"
+				+ "        \"Developer\",\n" + "        \"Engineer\"\n" + "    ],\n" + "    \"AITags\": null,\n"
+				+ "    \"versionInfo\": {\n" + "        \"id\": \"62affef97db937b028f3b47a\",\n"
+				+ "        \"name\": \"Version 1\"\n" + "    },\n" + "    \"embeddedMetadata\": {},\n"
+				+ "    \"customCoordinates\": null,\n" + "    \"customMetadata\": {},\n"
+				+ "    \"isPrivateFile\": false,\n"
+				+ "    \"url\": \"https://ik.imagekit.io/zv3rkhsym/sample-image11_g1Qv0wpqP.jpg\",\n"
+				+ "    \"thumbnail\": \"https://ik.imagekit.io/demo/img/static-file-1.png\",\n"
+				+ "    \"fileType\": \"non-image\",\n" + "    \"filePath\": \"/sample-image11_g1Qv0wpqP.jpg\",\n"
+				+ "    \"size\": 169170\n" + "}";
+		server.enqueue(new MockResponse().setBody(responseJson));
+		server.start();
+		RestClient.API_BASE_URL = server.url("/").toString();
+		SUT.getFileDetail("62affef97db937b028f3b47a");
+
+		RecordedRequest request = server.takeRequest();
+		String utf8RequestBody = request.getBody().readUtf8();
+		assertEquals("", utf8RequestBody);
+		assertEquals("application/json", request.getHeader("Content-Type"));
+		assertEquals("GET /v1/files/62affef97db937b028f3b47a/details HTTP/1.1", request.getRequestLine());
+		assertEquals(RestClient.API_BASE_URL.concat("v1/files/62affef97db937b028f3b47a/details"),
+				request.getRequestUrl().toString());
+	}
+
+	@Test
+	public void imageKit_getFileMetaData_successExpected()
+			throws IOException, ForbiddenException, TooManyRequestsException, InternalServerException,
+			UnauthorizedException, BadRequestException, UnknownException, InterruptedException {
+		MockWebServer server = new MockWebServer();
+		String responseJson = "{\n" + "    \"height\": 1000,\n" + "    \"width\": 1000,\n" + "    \"size\": 147022,\n"
+				+ "    \"format\": \"jpg\",\n" + "    \"hasColorProfile\": false,\n" + "    \"quality\": 0,\n"
+				+ "    \"density\": 72,\n" + "    \"hasTransparency\": false,\n" + "    \"exif\": {},\n"
+				+ "    \"pHash\": \"e0d52b612ad538f6\"\n" + "}";
+		server.enqueue(new MockResponse().setBody(responseJson));
+		server.start();
+		RestClient.API_BASE_URL = server.url("/").toString();
+		SUT.getRemoteFileMetadata("https://ik.imagekit.io/zv3rkhsym/default-image.jpg");
+
+		RecordedRequest request = server.takeRequest();
+		String utf8RequestBody = request.getBody().readUtf8();
+		assertEquals("", utf8RequestBody);
+		assertEquals("application/json", request.getHeader("Content-Type"));
+		assertEquals("GET /v1/metadata?url=https://ik.imagekit.io/zv3rkhsym/default-image.jpg HTTP/1.1",
```

Comment:
Do we need to handle any other cases where url is missing?

Reply:
It is already covered in the test case for 400 response code for this metadata endpoint.